src/boosting/gbdt_prediction.cpp

```diff
@@ -44,4 +44,25 @@ void GBDT::PredictLeafIndex(const double* features, double* output) const {
   }
 }
 
+void GBDT::PredictContrib(const double* features, double* output, const PredictionEarlyStopInstance* early_stop) const {
+  int early_stop_round_counter = 0;
+  // set zero
+  const int num_features = max_feature_idx_+1;
+  std::memset(output, 0, sizeof(double) * num_tree_per_iteration_ * (num_features+1));
+  for (int i = 0; i < num_iteration_for_pred_; ++i) {
+    // predict all the trees for one iteration
+    for (int k = 0; k < num_tree_per_iteration_; ++k) {
+      models_[i * num_tree_per_iteration_ + k]->PredictContrib(features, num_features, output + k*(num_features+1));
+    }
+    // check early stopping
+    ++early_stop_round_counter;
+    if (early_stop->round_period == early_stop_round_counter) {
```

Comment:
I think PredictContrib should do the same thing as PredictRaw, so that when people predict something and then explain it they are explaining the same prediction they just made.